exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/columnreaders/NullableVarLengthValuesColumn.java

```diff
@@ -69,11 +73,16 @@ protected boolean readAndStoreValueSizeInformation() throws IOException {
     if ( currDefLevel == -1 ) {
       currDefLevel = pageReader.definitionLevels.readInteger();
     }
-    if ( columnDescriptor.getMaxDefinitionLevel() > currDefLevel) {
+
+    if (columnDescriptor.getMaxDefinitionLevel() > currDefLevel) {
       nullsRead++;
-      // set length of zero, each index in the vector defaults to null so no need to set the nullability
-      variableWidthVector.getMutator().setValueLengthSafe(
-          valuesReadInCurrentPass + pageReader.valuesReadyToRead, 0);
+      // set length of zero, each index in the vector defaults to null so no
+      // need to set the nullability
+      if (variableWidthVector == null) {
```

Comment:
I know this class hierarchy is a bit messy as is ( I am the original author of most if it, I should go back to clean it up). But we shouldn't be using the presence or absence of this field as a flag to know which class we are in.

I'm